FEXCore/Source/Interface/IR/Passes/x87StackOptimizationPass.cpp

```diff
@@ -0,0 +1,1892 @@
+#include "FEXCore/Utils/LogManager.h"
+#include "Interface/IR/IR.h"
+#include "Interface/IR/IREmitter.h"
+#include "Interface/IR/PassManager.h"
+#include <FEXCore/IR/IR.h>
+#include <FEXCore/Utils/Profiler.h>
+#include <FEXCore/fextl/deque.h>
+
+#include <array>
+#include <cstddef>
+#include <cstdint>
+#include <stdint.h>
+
+namespace FEXCore::IR {
+
+// FixedSizeStack is a model of the x87 Stack where each element in this
+// fixed size stack lives at an offset from top. The top of the stack is at
+// index 0.
+template<typename T, size_t MaxSize>
```

Comment:
Why is this a template, instead of just adding a `const size_t MaxSize = 8` and hardcoding the type.

Reply:
Making it a template highlights that this is a generic container that doesn't peek into any of `StackMemberInfo`'s internal state. This is a simple template use so I don't see much of a downside.

Reply:
fair, I don't feel strongly on this one either way.